### ISSUE TITLE
Unable to override controller methods with Authorize DataAnnotation

### ISSUE THREAD

**Issue by sc14aec** [Research-Institute/json-api-dotnet-core#26]
We have JWT authentication middleware configured on our application - so any controllers with the [Authorize] data annotation will return 401 Unauthorized unless a valid token is provided in the request header.

We tried to apply the data annotation to an endpoint as follows:

```csharp
    [Authorize]
    public class ReadingController : JsonApiController, IJsonApiController
    {
        private dbContext _context;

        public ReadingController(JsonApiContext jsonApiContext, ResourceRepository resourceRepository,
            dbContext dbContext) : base(jsonApiContext, resourceRepository)
        {
            _context = dbContext;
        }

        [Authorize]
        public override ObjectResult Get()
        {
            return base.Get();
        }

        [Authorize]
        public override ObjectResult Get(string id)
        {
            return base.Get(id);
        }
    }
```

But the endpoint returns a 200 OK response with the data even when no token is provided

**Comment by jaredcnance** [comment]
Thanks for the feedback. Currently, this is the expected behavior. The current design bypasses all aspnet MVC middleware that would be used as part of the `app.AddMvc()` hook, including the auth hooks prior to activating the controller. I'd like to figure out how to lean on the built in mechanisms more. This is the main reason I haven't worked on this project in a few weeks. I'm spending free time researching and messing around with better approaches.

**Comment by jaredcnance** [comment]
Just as an update, I am still working on this. This is a pretty major change and will result in deprecation of this repo. The new version works WITH the MVC routing/controller middleware so authentication will still be enabled.

**Comment by jaredcnance** [comment]
This has now been fixed in the latest version on NuGet (0.2.7). Please take a look at the Readme as the usage has gone through significant changes.
